peps/pep-0788.rst

```diff
@@ -701,48 +916,90 @@ For example:
     static PyObject *
     setup_callback(PyObject *self, PyObject *unused)
     {
-        PyThread_handle_t handle;
-        PyThead_indent_t indent;
-
-        pyrun_t *data = PyMem_RawMalloc(sizeof(pyrun_t));
-        if (data == NULL) {
-            return PyErr_NoMemory();
-        }
         // Weak reference to the interpreter. It won't wait on the callback
         // to finalize.
-        data->interp_id = PyInterpreterState_GetID(PyInterpreterState_Get());
-        register_callback(async_callback, data);
+        PyInterpreterWeakRef wref = PyInterpreterWeakRef_Get();
+        register_callback(async_callback, wref);
 
         Py_RETURN_NONE;
     }
 
+Example: Calling Python Without a Closure
+*****************************************
+
+There are a few cases where callback functions don't take a closure
+(``void *arg``), so it's impossible to acquire a reference to any specific
```

Comment:
A closure is a function that captures its environment. The combined callback and argument behave kind of like a closure, but calling the `void *arg` by itself a "closure" seems strange to me. I'd call it a "callback parameter" or "callback argument"